### ISSUE TITLE
Unfortunately, an error has occurred

### ISSUE THREAD

**Issue by ErnDevDDLC** [ScottMichaud/YandereSimulatorLauncher2#33]
I wanted to download yandere simulator, but when I start the download it gives me an error.

![image](https://user-images.githubusercontent.com/84817953/231874786-5e6abd9a-f8a8-4bb4-98cc-e82b638d4971.png)

I tried 3 days ago and today I tried again and it is still failing, maybe change the server where the launcher downloads the game (example: from mega.nz to mediafire), I hope it get fixed soon.

**Comment by ScottMichaudLt** [comment]
This is usually because the launcher is being run from inside the zip. This runs the launcher as a temporary file, and the unzip utility tells it to install in weird places. The one built into Windows tells it that its working directory is inside the System32 folder, so it cannot write the game there and fails.

Extract YandereSimLauncher.zip before running YandereSimulatorLauncher2.exe.

- Right-click on YandereSimLauncher.zip
- Click "Extract All"
- Rename the folder that Windows wants to create from YandereSimLauncher to YandereSimLauncherExtract (so you know which one's the zip and which one's the extracted folder).
- Open the YandereSimLauncherExtract folder that was just created.
- Run YandereSimulatorLauncher.exe from that new YandereSimLauncherExtract folder.

This is intended as a feature because it lets users install the game onto USB drives or anywhere they put the launcher (or point a shortcut to, because those can also change the working directory).

**Comment by ErnDevDDLC** [comment]
> 

But I'm not running the launcher from the .zip, so that's not it, I also made sure that the folder does not have read-only permission enabled.

**Comment by ScottMichaudLt** [comment]
When the launcher is running, could you open Task Manager, right-click on YandereSimulatorLauncher2.exe, and click on "Open File Location". A Windows Explorer folder will open. Where is this folder located?

**Comment by ErnDevDDLC** [comment]
> When the launcher is running, could you open Task Manager, right-click on YandereSimulatorLauncher2.exe, and click on "Open File Location". A Windows Explorer folder will open. Where is this folder located?

C:\Games\Yandere Simulator

**Comment by ScottMichaudLt** [comment]
Okay so in that folder is YandereSimulatorLauncher2.exe and you are running the .exe directly (no shortcuts)?

**Comment by ErnDevDDLC** [comment]
> Okay so in that folder is YandereSimulatorLauncher2.exe and you are running the .exe directly (no shortcuts)?

Yeah

**Comment by ScottMichaudLt** [comment]
Hmm. Could you place YandereSimulatorLauncher2.exe on your Desktop and run it there (as a test for a file that 100% is writable to your user)? The error is a catch-all, but it's code that hasn't been changed since 2020, and it continues to work here so it would be really weird to have one random failure 3 years in. (It's a third-party library that specifically downloads Mega.nz links.)

In general I've seen it occur with either a network issue going to Mega.nz, or it cannot write the file.

One user who had this problem was trying to connect to Mega.nz with a mobile hotspot. What sort of internet connection do you have?

**Comment by ErnDevDDLC** [comment]
> Hmm. ¿Podría colocar YandereSimulatorLauncher2.exe en su escritorio y ejecutarlo allí (como una prueba para un archivo que 100% es grabable para su usuario)? El error es un cajón de sastre, pero es un código que no se ha cambiado desde 2020, y continúa funcionando aquí, por lo que sería realmente extraño tener una falla aleatoria 3 años después. (Es una biblioteca de terceros que descarga específicamente Mega.nz enlaces).
> 
> En general, he visto que ocurre con un problema de red que va a Mega.nz o no puede escribir el archivo.
> 
> Un usuario que tenía este problema intentaba conectarse a Mega.nz con un punto de acceso móvil. ¿Qué tipo de conexión a Internet tienes?

Ethernet

**Comment by ErnDevDDLC** [comment]
> Hmm. Could you place YandereSimulatorLauncher2.exe on your Desktop and run it there (as a test for a file that 100% is writable to your user)? The error is a catch-all, but it's code that hasn't been changed since 2020, and it continues to work here so it would be really weird to have one random failure 3 years in. (It's a third-party library that specifically downloads Mega.nz links.)
> 
> In general I've seen it occur with either a network issue going to Mega.nz, or it cannot write the file.
> 
> One user who had this problem was trying to connect to Mega.nz with a mobile hotspot. What sort of internet connection do you have?

I have ethernet connection, and even if I put the .exe on the desktop it still doesn't work.

**Comment by ScottMichaudLt** [comment]
Can you download the game from Mega.nz through the Wordpress site?

https://yanderedev.wordpress.com/downloads/

I realize that I misread the original post. I thought that `maybe change the server where the launcher downloads the game (example: from mega.nz to mediafire)` meant "maybe YandereDev changed the server and you need to update" but you might have been asking for us to provide an alternative to Mega.nz.

**Comment by ErnDevDDLC** [comment]
> Can you download the game from Mega.nz through the Wordpress site?
> 
> https://yanderedev.wordpress.com/downloads/
> 
> I realize that I misread the original post. I thought that `maybe change the server where the launcher downloads the game (example: from mega.nz to mediafire)` meant "maybe YandereDev changed the server and you need to update" but you might have been asking for us to provide an alternative to Mega.nz.

I can download the game from mega.nz, but from the browser, from the launcher I can't, and don't worry because you misread the post, that usually happens.

**Comment by ScottMichaudLt** [comment]
Yeah not sure on this one. The only thing that would make sense to me is if you're using a web browser with a built-in VPN, but the launcher is getting blocked when attempting to directly connect.

I'm likely wrong of course, but that's the only thing that I can think of at the moment.

It would be nice to support multiple download sources, and have a tighter algorithm for handling where to install the game (ex: leave an install location in appdata, but load the version next to the launcher EXE if it exists). Unfortunately I haven't really touched the launcher in about a year. (Job changes and such.)

Of course the game could be downloaded from here: https://yanderedev.wordpress.com/downloads/ -- the launcher only really tells the user when a new version is available. Frankly it would probably have been better to do that in-game, anyway.

**Comment by ErnDevDDLC** [comment]
> Yeah not sure on this one. The only thing that would make sense to me is if you're using a web browser with a built-in VPN, but the launcher is getting blocked when attempting to directly connect.
> 
> I'm likely wrong of course, but that's the only thing that I can think of at the moment.
> 
> It would be nice to support multiple download sources, and have a tighter algorithm for handling where to install the game (ex: leave an install location in appdata, but load the version next to the launcher EXE if it exists). Unfortunately I haven't really touched the launcher in about a year. (Job changes and such.)
> 
> Of course the game could be downloaded from here: https://yanderedev.wordpress.com/downloads/ -- the launcher only really tells the user when a new version is available. Frankly it would probably have been better to do that in-game, anyway.

I don't got a vpn so it's weird, anyway

Thanks

**Comment by Sophiaaaaaaxd** [comment]
Okay, so, the problem is that, the game needed an update, tried the launcher, didn't work, so i downloaded the update from one of the websites. Still didn't work, tried launcher again, didn't work once again, deleted the whole game, downloaded it once again, from one of the websites, because the launcher still didn't work, but now the launcher is giving me only the 'install' option, and i also cannot open the game somehow.  
![Képernyőkép 2024-03-02 023525](https://github.com/ScottMichaud/YandereSimulatorLauncher2/assets/161904554/ef601b9b-a2f4-43d2-a60b-eee48dc78b61)

**Comment by CloudDutch** [comment]
I also had this issue with downloading from the launcher, but before that when I downloaded from a fixed network on my computer in my hometown, it worked fine. Now that I'm out to work, I'm downloading through the apartment's WiFi connection, and I'm getting an error
